secp256k1_test.go

```diff
@@ -501,6 +501,126 @@ func TestDeterministicSchnorrSignatureGen(t *testing.T) {
+	}
+
+	if !nilPubKey.IsEqual(nil) {
+		t.Errorf("two nil pubkeys should be equal")
+	}
+
+	copyGoodPubkey := *goodPublicKey
+	if !copyGoodPubkey.IsEqual(goodPublicKey) {
+		t.Errorf("A pubkey and it's copy should be the same")
+	}
+	goodPrivateKey2 := fastGeneratePrivateKey(t, r)
+	goodPublicKey2, err := goodPrivateKey2.SchnorrPublicKey()
+	if err != nil {
+		t.Fatalf("Failed generating pubkey from: '%s'. '%s'", goodPrivateKey2, err)
+	}
+
+	if goodPublicKey.IsEqual(goodPublicKey2) {
+		t.Errorf("'%s' shouldn't be equal to %s", goodPublicKey, goodPublicKey2)
+	}
+}
+
+func TestSchnorrSignature_IsEqual(t *testing.T) {
+	r := rand.New(rand.NewSource(1))
+	serializedSig := SerializedSchnorrSignature{}
+	n, err := r.Read(serializedSig[:])
+	if err != nil || n != len(serializedSig) {
+		t.Errorf("Failed generating a random signature '%d' '%s' ", n, err)
+	}
+	goodSignature := DeserializeSchnorrSignature(&serializedSig)
+
+	emptySignature := SchnorrSignature{}
+	if emptySignature.IsEqual(goodSignature) {
+		t.Errorf("Emtpy signature shouldn't be equal to good one")
+	}
+	if !emptySignature.IsEqual(&SchnorrSignature{}) {
+		t.Errorf("Emtpy signature should be equal to another empty signature")
+	}
+	var nilSignature *SchnorrSignature = nil
+	if nilSignature.IsEqual(goodSignature) {
+		t.Errorf("nil signature shouldn't be equal to good one")
+	}
+
+	if !nilSignature.IsEqual(nil) {
+		t.Errorf("two nil signatures should be equal")
+	}
+
+	copyGoodSignature := *goodSignature
+	if !copyGoodSignature.IsEqual(goodSignature) {
+		t.Errorf("A signature and it's copy should be the same")
+	}
+
+	serializedSig2 := SerializedSchnorrSignature{}
+	n, err = r.Read(serializedSig2[:])
+	if err != nil || n != len(serializedSig2) {
+		t.Errorf("Failed generating a random signature '%d' '%s' ", n, err)
+	}
+
+	goodSignature2 := DeserializeSchnorrSignature(&serializedSig2)
+	if goodSignature.IsEqual(goodSignature2) {
+		t.Errorf("'%s' shouldn't be equal to %s", goodSignature, goodSignature2)
+	}
+}
+
+func TestHash_IsEqual(t *testing.T) {
+	r := rand.New(rand.NewSource(1))
+	goodHash := &Hash{}
+	n, err := r.Read(goodHash[:])
+	if err != nil || n != len(goodHash) {
+		t.Errorf("Failed generating a random hash '%d' '%s' ", n, err)
+	}
+
+	emptyHash := Hash{}
+	if emptyHash.IsEqual(goodHash) {
+		t.Errorf("Emtpy hash shouldn't be equal to filled one")
+	}
+	if !emptyHash.IsEqual(&Hash{}) {
+		t.Errorf("Emtpy hash should be equal to another empty hash")
+	}
+	var nilHash *Hash = nil
+	if nilHash.IsEqual(goodHash) {
+		t.Errorf("nil hash shouldn't be equal to good one")
+	}
+
+	if !nilHash.IsEqual(nil) {
+		t.Errorf("two nil hashes should be equal")
+	}
+
+	copyGoodHash := *goodHash
+	if !copyGoodHash.IsEqual(goodHash) {
+		t.Errorf("A hash and it's copy should be the same")
+	}
+
+	goodHash2 := &Hash{}
+	n, err = r.Read(goodHash2[:])
+	if err != nil || n != len(goodHash2) {
+		t.Errorf("Failed generating a random hash '%d' '%s' ", n, err)
```

Comment:
Please annotate what n is in the error and remove the whitespace at the end of the line.

Reply:
Same as above